src/test/java/beam/analysis/plots/DeadHeadingGraphTest.java

```diff
@@ -9,7 +9,7 @@
 import static org.junit.Assert.assertEquals;
 
 public class DeadHeadingGraphTest {
-    private final DeadHeadingAnalysis deadHeadingStats = new DeadHeadingAnalysis(NoOpSimulationMetricCollector$.MODULE$, true);
+    private final DeadHeadingAnalysis deadHeadingStats = new DeadHeadingAnalysis(NoOpSimulationMetricCollector$.MODULE$, true, null);
```

Comment:
nulls scare me a bit. Should we make this an Optional at least - or fill it with a mocked object?

Reply:
I created a mock for this test. At least it provides a meaningful exeception instead of an NPE if it is called.